railties/lib/rails/commands/console/console_command.rb

```diff
@@ -12,6 +12,47 @@ def filter_backtrace(bt)
       end
     end
 
+    class IRBConsole
+      def initialize
+        require "irb"
+        require "irb/completion"
+        IRB::WorkSpace.prepend(BacktraceCleaner)
+
+        if !Rails.env.local?
+          # Use env var here so users can override them with env var too
+          ENV["IRB_USE_AUTOCOMPLETE"] ||= "false"
+        end
+
+        IRB
+      end
+
+      def name
+        "IRB"
+      end
+
+      def start
+        IRB.setup(nil)
+
+        env =
+          if Rails.env == "development"
+            "dev"
+          else
+            Rails.env.first(4)
+          end
+
+        IRB.conf[:PROMPT][:RAILS_PROMPT] = {
+          PROMPT_I: "#{env}:%03n> ",
+          PROMPT_S: "#{env}:%03n%l ",
+          PROMPT_C: "#{env}:%03n* ",
+          RETURN: "=> %s\n"
+        }
+
+        # Respect user's choice of prompt mode.
+        IRB.conf[:PROMPT_MODE] = :RAILS_PROMPT if IRB.conf[:PROMPT_MODE] == :DEFAULT
```

Comment:
Would it be easy to add a test for this behavior ("Respect user's choice of prompt mode")?

Reply:
Added 👍